lib/scope.js

```diff
@@ -75,33 +75,32 @@ module.exports = {
     return trimmed;
   },  // Ends scope.toTrimmedJSON()
 
-  setReportHeading: async function (scope, { scenario }) {
+  addReportHeading: async function (scope, { scenario }) {
+    /** Return printable heading based on Scenario name and tags.
+    *    Allow characters from other languages.
+    */
     if ( env_vars.DEBUG ) { console.log( scenario.pickle.name ); }
-    // Names for files and reports
-    // TODO: Allow scenario ids to have digits in their name
 
-    // With tag names automatically added to langauges, this should work to make
-    // each language unique.
-    let tags = '';
+    // Collect all tag names
+    let tag_names = [];
     for ( let tag of scenario.pickle.tags ) {
-      tags += tag.name.replace(/[^A-Za-z0-9]+/g, '_').replace(/(\P{L})+/gu, '_');
+      tag_names.push( tag.name );
     }
-    tags = tags.replace(/_$/, '');
-    let new_id = `${ scenario.pickle.name }${ tags }`;
 
-    // Should we add a counter? It would have to be unique to each scenario name.
-    // Not sure of another good way to indicate the _name_ is a duplicate
-    while ( scope.report.get( new_id ) !== undefined  ) {
-      new_id += '_';
+    let heading = `\n---------------\n`
+      + `Scenario: ${ scenario.pickle.name }`;
+
+    if ( tag_names.length > 0 ) {
+      heading += `\nTags: ${ tag_names.join(' ') }`;
     }
```

Comment:
```suggestion
    if ( tag_names ) {
      heading += `\nTags: ${ tag_names.join(' ') }`;
    }
```

Reply:
I think you said this is a convention in some linters. You also said you prefer the way the code currently is. Thanks for bringing up the discussion. When we talk about code style, we can include this as one of the items. For now, I, like you, prefer to be explicit.

Reply:
I double-checked, apparently, my suggestion is incorrect for JavaScript. 
```
if ( tag_names.length ) {
``` should be possible but it doesn't change much X)